src/Checkbox/Checkbox.tsx

```diff
@@ -81,11 +81,14 @@ export default class Checkbox extends React.PureComponent<Props> {
     if (ariaCheckedState && partial) {
       ariaCheckedState = "mixed";
     }
+    const hasLabel = !!children;
```

Comment:
Checkboxes with children use the child as the aria-labelled by. In cases where the checkbox has no child, there is nothing to act as an aria-label